pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -316,7 +316,7 @@ def model_sharded_context(self) -> Generator[None, None, None]:
         else:
             model_parallel_context = super().model_sharded_context()
 
-        with model_parallel_context:
+        with torch.cuda.amp.autocast(), model_parallel_context:
```

Comment:
I think for ZeRO 2/3, autocast isn't used as we rely on a custom DeepSpeed FP16Optimizer to handle this, which converts the entire model to half iirc.

Reply:
With newer DeepSpeed version, they check the weights are created in half precision and currently fails in CI.

Reply:
oh very interesting! i'll check myself when I get time, do they throw an error if the weights are in half precision or not in half precision?